src/library/scala/collection/mutable/TreeSet.scala

```diff
@@ -104,6 +104,13 @@ sealed class TreeSet[A] private (tree: RB.Tree[A, Null])(implicit val ordering:
   override def last = RB.maxKey(tree).get
   override def lastOption = RB.maxKey(tree)
 
+
+  override def min[B >: A](implicit cmp: Ordering[B]): A =
+    if ((cmp eq ordering) && nonEmpty) head else super.min(cmp)
```

Comment:
none of the instances provided by the 2.12 standard library override `equals` (yet?), but there is not in principle a reason someone could not implement it for their own instance(s). consequently, using `==` (here and below) might be more appropriate. on the other hand, maybe it's not worth the effort

Reply:
Thank you for the review. It seems like it's not worth the effort for me.  It is my first PR here 😊  and I do not want to complicate it. Could you please elaborate on your idea about overriding `equals` for own instance? What do you mean by "own instance" ?

Reply:
"instance" was the wrong word. I meant implementation. e.g. for `case class Foo[A](a: A, int: Int)`, you could implement an `Ordering` that overrides `equals`, as the implementations in 2.13 do

Reply:
I have updated the PR (including immutable.TreeSet)